keel-plugin/src/main/kotlin/com/netflix/spinnaker/keel/plugin/ResourceHandler.kt

```diff
@@ -15,155 +15,15 @@
-   */
-  fun generateName(spec: T): ResourceName
-
-  /**
-   * Validates the resource and applies any defaults / normalization.
-   *
-   * @return [resource] or a copy of [resource] that may have been changed, for example in order to
-   * set default values.
-   */
-  @Suppress("UNCHECKED_CAST")
-  fun normalize(resource: Resource<Any>): Resource<T> {
-    var normalizedResource = resource.copy(
-      spec = objectMapper.convertValue(resource.spec, supportedKind.second)
-    ) as Resource<T>
-    for (normalizer in normalizers) {
-      if (normalizer.handles(resource.apiVersion, resource.kind)) {
-        log.debug("Normalizing ${resource.metadata.name} with ${normalizer.javaClass}")
-        normalizedResource = normalizer.normalize(normalizedResource) as Resource<T>
-      }
-    }
-    return normalizedResource
-  }
-
-  /**
-   * Return the current _actual_ representation of what [resource] looks like in the cloud.
-   * The entire desired state is passed so that implementations can use whatever identifying
-   * information they need to look up the resource. Implementations of this method should not
-   * actuate any changes.
-   */
-  fun current(resource: Resource<T>): T?
-
-  /**
-   * Create a resource so that it matches the desired state represented by [resource].
-   *
-   * By default this just delegates to [upsert].
-   *
-   * Implement this method and [update] if you need to handle create and update in different ways.
-   * Otherwise just implement [upsert].
-   *
-   * @return a list of tasks launched to actuate the resource.
-   */
-  fun create(resource: Resource<T>): List<TaskRef> =
-    upsert(resource, null)
-
-  /**
-   * Update a resource so that it matches the desired state represented by [resource].
-   *
-   * By default this just delegates to [upsert].
-   *
-   * Implement this method and [create] if you need to handle create and update in different ways.
-   * Otherwise just implement [upsert].
-   *
-   * @return a list of tasks launched to actuate the resource.
-   */
-  fun update(
-    resource: Resource<T>,
-    resourceDiff: ResourceDiff<T> = ResourceDiff(resource.spec)
-  ): List<TaskRef> =
-    upsert(resource, resourceDiff)
-
-  /**
-   * Create or update a resource so that it matches the desired state represented by [resource].
-   *
-   * You don't need to implement this method if you are implementing [create] and [update]
-   * individually.
-   *
-   * @return a list of tasks launched to actuate the resource.
-   */
-  fun upsert(resource: Resource<T>, resourceDiff: ResourceDiff<T>? = null): List<TaskRef> {
-    TODO("Not implemented")
-  }
+interface ResourceHandler<T : Any> : ResolvableResourceHandler<T, T> {
 
   /**
-   * Delete a resource as the desired state is that it should no longer exist.
+   * This method is a no-op when the model and resolved types are the same.
    */
-  fun delete(resource: Resource<T>)
+  override fun desired(resource: Resource<T>): T = resource.spec
```

Comment:
In the `ResourceHandler` interface spec type and current/desired type are the same.